### PR TITLE
Add various tests, add RNG_RANDOM_TARGET

### LINKED ISSUE #5445: Uproar test potentially wrong
### Description

The uproar test (https://github.com/rh-hideout/pokeemerald-expansion/blob/master/test/battle/move_effect/uproar.c#L9) assigns Voltorb soundproof but also expects it to wake up from soundproof, which is incorrect behavior according to <https://github.com/rh-hideout/pokeemerald-expansion/blob/1c022f4d34ef4fdf6a894815fd4b4617b15932ed/src/battle_util.c#L2664>. 

Conversation/GIF starts here: https://discord.com/channels/419213663107416084/1077009799293710357/1290119643612184718

### Version

1.9.2 (Latest release)

### Upcoming/master Version

_No response_

### Discord contact info

_No response_

### PR DESCRIPTION
**Pull request by ghoulslash**
This is kind of a running list of new tests with no particular strategy. It is not a draft but is probably a 'living' pull request. Can be merged at anytime, though.

Tests included:
- B_BURN_DAMAGE will change the test to 1/16th or 1/8th damage accordingly
- Add Belly Drum + Contrary Test 
- Update kings shield test with stat change value config
- Sticky web + contrary test

Also Adds:
- RNG_RANDOM_TARGET and cleans up some uproar/random target code. Closes #5445

### CODE REVIEW

File: test/battle/status1/burn.c [hunk 1]
```diff
@@ -17,6 +17,23 @@ SINGLE_BATTLE_TEST("Burn deals 1/16th damage per turn")
             HP_BAR(player, damage: maxHP / 16);
     }
 }
+#else
+SINGLE_BATTLE_TEST("Burn deals 1/8th damage per turn")
```
Comment (AlexOn1ine) [inline test/battle/status1/burn.c:21]: Instead of preproc'ing you could do and if else inside the first statement depending on the gen

Comment (Bassoonian) [reply test/battle/status1/burn.c:21]: We could also just use a ternary for 16 : 8. That way, only the title needs to be checked

Comment (ghoulslash) [reply test/battle/status1/burn.c]: resolved with ternary. good suggestion

File: src/battle_util.c [hunk 2]
```diff
@@ -251,20 +251,7 @@ void HandleAction_UseMove(void)
...
-                        gBattlerTarget = GetBattlerAtPosition(B_POSITION_OPPONENT_RIGHT);
-                }
-                else
-                {
-                    if (Random() & 1)
-                        gBattlerTarget = GetBattlerAtPosition(B_POSITION_PLAYER_LEFT);
-                    else
-                        gBattlerTarget = GetBattlerAtPosition(B_POSITION_PLAYER_RIGHT);
-                }
+                gBattlerTarget = SetRandomTarget(gBattlerAttacker);
```
Comment (AlexOn1ine) [inline src/battle_util.c:254]: any reason not to use `gBattlerAttacker` directly instead of passing it?

Comment (AlexOn1ine) [reply src/battle_util.c:254]: nvm. I didn't realize it is used somewhere else sorry

File: src/battle_script_commands.c [hunk 3]
```diff
@@ -11476,10 +11476,11 @@ static void Cmd_setmiracleeye(void)
 bool8 UproarWakeUpCheck(u8 battler)
 {
     s32 i;
+    bool32 hasSoundproof = (GetBattlerAbility(battler) == ABILITY_SOUNDPROOF && B_UPROAR_IGNORE_SOUNDPROOF < GEN_5);
```
Comment (Bassoonian) [inline src/battle_script_commands.c:11479]:
```suggestion
    bool32 hasSoundproof = (B_UPROAR_IGNORE_SOUNDPROOF < GEN_5 && GetBattlerAbility(battler) == ABILITY_SOUNDPROOF);
```

If I'm not mistaken, leading with the config means that the compiler can optimise away the check that follows it

File: src/battle_util.c [hunk 4]
```diff
@@ -2662,7 +2635,7 @@ u8 DoBattlerEndTurnEffects(void)
                 for (gBattlerAttacker = 0; gBattlerAttacker < gBattlersCount; gBattlerAttacker++)
                 {
                     if ((gBattleMons[gBattlerAttacker].status1 & STATUS1_SLEEP)
-                     && GetBattlerAbility(gBattlerAttacker) != ABILITY_SOUNDPROOF)
+                     && ((GetBattlerAbility(gBattlerAttacker) != ABILITY_SOUNDPROOF) || B_UPROAR_IGNORE_SOUNDPROOF >= GEN_5))
```
Comment (Bassoonian) [inline src/battle_util.c:2638]:
```suggestion
                     && ((B_UPROAR_IGNORE_SOUNDPROOF >= GEN_5) || (GetBattlerAbility(gBattlerAttacker) != ABILITY_SOUNDPROOF))
```

Same as above

File: test/battle/move_effect/uproar.c [hunk 5]
```diff
@@ -8,6 +8,7 @@ ASSUMPTIONS
 
 DOUBLE_BATTLE_TEST("Uproar status causes sleeping pokemon to wake up during an attack")
 {
+    PASSES_RANDOMLY(1,2,RNG_RANDOM_TARGET); // test fails if we target soundproof mon
```
Comment (Bassoonian) [inline test/battle/move_effect/uproar.c:11]:
```suggestion
    PASSES_RANDOMLY(1, 2, RNG_RANDOM_TARGET); // test fails if we target soundproof mon
```

**Review by AlexOn1ine — APPROVED** [review]

**Pull request merged**